### DISCUSSION TITLE
리액트 프로젝트에서 테스트는 어떻게 하나요?

### DISCUSSION THREAD

**Discussion by airman5573** [airman5573/fe-20s-interview#229] [Q&A]
-

**Comment by cksal0805** [comment]
리액트 프로젝트는 유틸함수, 컴포넌트, E2E 테스트를 보통 진행하는데, 여러가지 방법론과 도구를 사용할 수 있어요.
방법론에는 TDD, BDD등을 사용할 수 있고, 도구로는

- 유틸함수들을 테스트? vitest 혹은 jest 를 활용해서 단위 테스트를 진행 할 수 있어요.
- 컴포넌트 테스트, 컴포넌트 통합 테스트? happy-dom, @testing-library/react로 리액트 컴포넌트 테스트, msw를 활용해 api 응답 mocking 환경을 구축 하여 테스트를 진행할 수 있어요.
- E2E 테스트 ? playwright 활용 하여 live 혹은 alpha환경에서테스트 진행 할 수 있어요.
